### ISSUE TITLE
VMware: Provide a `cargo make` target for OVA upload/template

### ISSUE THREAD

**Issue by zmrow** [bottlerocket-os/bottlerocket#1605]
**Feature**
OVAs are already automatically built for VMware variants.  It would be nice to provide a friendly `cargo make` target to upload OVAs to one or more vSphere datacenters, optionally making them into VM templates.

**`cargo make` interface**
We will provide two new make targets `cargo make upload-ova` and `cargo make vmware-template`.  Providing the `upload-ova` target means that one can upload an OVA without making it a template, which can also be useful for some users.  The `vmware-template` target will upload the OVA automatically so a user doesn't need to run multiple targets.

**Under the hood**
We will build `govc` into the SDK container, which a user will already have if they have built any images.  `pubsys` (called via `cargo make`) will orchestrate `docker run` commands to call `govc`.  `govc` configuration will be provided via environment variables in the `docker run` call.  Configuration will be provided for 1 or more datacenters in `Infra.toml` and a new file `~/.config/pubsys/vsphere-credentials.toml`.

At runtime, `pubsys` will iterate through all the datacenters provided in `Infra.toml`.  A user can upload to a subset of datacenters using an environment variable.

Users can provide configuration for multiple datacenters in their own TOML blocks (see below).  Any "common" configuration can be provided for keys that are the same for all datacenters.  **_All configuration may be overridden at runtime via `GOVC_` environment variables_**.  

Credentials will be provided in `~/.config/pubsys/vsphere-credentials.toml`.  At runtime, `pubsys` will look in this file for credentials for the current datacenter.  If they do not exist, it will look for the environment variables `GOVC_USERNAME` and `GOVC_PASSWORD`.

**Infra.toml**
```
[vmware]
datacenters = ["foo", "bar", "baz"]

# Optional common configuration.
# This configuration allow values to be set in a single place if they are common in 
# multiple datacenters.  They can be overridden in the datacenter's block
[vmware.common]
network = "a_network"

[vmware.datacenter.foo]
vsphere_url = "https://a_datacenter.com" # GOVC_URL
datacenter = "a_datacenter" # GOVC_DATACENTER
datastore = "a_datastore" # GOVC_DATASTORE
network = "another_network" # GOVC_NETWORK
folder = "a_folder" # GOVC_FOLDER
resource_pool = "/a/resource/pool" # GOVC_RESOURCE_POOL

[vmware.datacenter.bar]
...
```

**vsphere-credentials.toml**
```
[datacenter.foo]
username = "username" # GOVC_USERNAME
password = "password" # GOVC_PASSWORD

[datacenter.bar]
username = "bar"
password = "baz"
```

**Import spec**
The following import spec will be used when uploading OVAs to vSphere.  Appropriate values will be substituted in the template for `Network` and `markastemplate`.  A user may override this spec with their own file using an environment variable.
```
{
  "DiskProvisioning": "flat",
  "IPAllocationPolicy": "dhcpPolicy",
  "IPProtocol": "IPv4",
  "NetworkMapping": [
    {
      "Name": "VM Network",
      "Network": "{{network}}"
    }
  ],
  "MarkAsTemplate": {{mark_as_template}},
  "PowerOn": false,
  "InjectOvfEnv": false,
  "WaitForIP": false,
  "Name": null
}
```

**Comment by jpculp** [comment]
https://github.com/bottlerocket-os/bottlerocket-sdk/pull/54 is merged, but `govc` won't be available to build with until a new SDK is released.

**Comment by zmrow** [comment]
Implemented in #1622
